Change routine load task sheduler interval to 0

EmmyMiao87 opened the pull request:
1. change routine load task scheduler interval to 0
2. init progress when routine load scheduler
3. add unit test and function test of routine load scheduler and task commit
4. login of rescheduler when kafka partition has changed

EmmyMiao87 committed:
Change routine load task sheduler interval to 0
1. change routine load task scheduler interval to 0
2. init progress when routine load scheduler
3. add unit test and function test of routine load scheduler and task commit

morningman commented on fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadManager.java:
```diff
@@ -402,4 +396,10 @@ public void removeOldRoutineLoadJobs() {
         // TODO(ml): remove old routine load job
     }
 
+    public void reSchedulerRoutineLoadJob() {
```
reSchedulerRoutineLoadJob  -> rescheduleRoutineLoadJob

morningman commented on fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadTaskScheduler.java:
```diff
@@ -28,52 +28,74 @@
+                return;
+            }
+
+            if (clusterIdleSlotNum > 0) {
```
You don't need to check if clusterIdleSlotNum > 0 here. Because if clusterIdleSlotNum <= 0, you will not enter this while loop at all. And you already make sure that clusterIdleSlotNum is always larger than needScheduledTaskNum.

morningman commented on fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadTaskScheduler.java:
```diff
@@ -28,52 +28,74 @@
+                    return;
                 }
                 RoutineLoadTask routineLoadTask = routineLoadTaskInfo.createStreamLoadTask(beId);
                 if (routineLoadTask != null) {
```
createStreamLoadTask seems never return null?

EmmyMiao87 commented on fe/src/main/java/org/apache/doris/load/routineload/KafkaRoutineLoadJob.java:
```diff
@@ -206,24 +235,33 @@ public static KafkaRoutineLoadJob fromCreateStmt(CreateRoutineLoadStmt stmt) thr
+        }
+        for (Integer kafkaPartition : currentKafkaPartitions) {
+            try {
+                ((KafkaProgress) progress).getPartitionIdToOffset().get(kafkaPartition);
```
Update the progress of new partitions

EmmyMiao87 commented on fe/src/main/java/org/apache/doris/load/routineload/KafkaRoutineLoadJob.java:
```diff
@@ -206,24 +235,33 @@ public static KafkaRoutineLoadJob fromCreateStmt(CreateRoutineLoadStmt stmt) thr
+            // fetch all of kafkaPartitions in topic
+            currentKafkaPartitions.addAll(getAllKafkaPartitions());
+        } else {
+            currentKafkaPartitions = customKafkaPartitions;
```
I will check it in setOptional()

EmmyMiao87 committed:
Add checker of custom kafka partition
1. need scheduler to need schedule
2. add checker of custom kafka partition when create routine load job
3. fix unit test error
